ethereum/spec/src/test/java/tech/pegasys/teku/spec/logic/common/statetransition/epoch/AbstractEpochProcessorTest.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright Consensys Software Inc., 2023
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.spec.logic.common.statetransition.epoch;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.lang.reflect.Field;
+import org.apache.logging.log4j.Logger;
+import org.junit.jupiter.api.Test;
+import org.junit.platform.commons.util.ReflectionUtils;
+import org.junit.platform.commons.util.ReflectionUtils.HierarchyTraversalMode;
+import tech.pegasys.teku.infrastructure.ssz.collections.SszPrimitiveVector;
+import tech.pegasys.teku.infrastructure.ssz.primitive.SszUInt64;
+import tech.pegasys.teku.infrastructure.time.Throttler;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.spec.Spec;
+import tech.pegasys.teku.spec.TestSpecFactory;
+import tech.pegasys.teku.spec.datastructures.state.Validator;
+import tech.pegasys.teku.spec.datastructures.state.beaconstate.BeaconState;
+import tech.pegasys.teku.spec.logic.versions.capella.statetransition.epoch.EpochProcessorCapella;
+import tech.pegasys.teku.spec.util.DataStructureUtil;
+
+class AbstractEpochProcessorTest {
+
+  private final Spec spec = TestSpecFactory.createMinimalCapella();
+  private final DataStructureUtil dataStructureUtil = new DataStructureUtil(spec);
+  private final EpochProcessorCapella epochProcessor =
+      (EpochProcessorCapella) spec.getGenesisSpec().getEpochProcessor();
+
+  @Test
+  public void shouldThrottleInactivityLeakLogs() throws Exception {
+    final int throttlingPeriod = 1; // expect maximum of one call per epoch
+    final Logger logger = mock(Logger.class);
+    final Throttler<Logger> loggerThrottler = spyLogThrottler(logger, throttlingPeriod);
+
+    final BeaconState state = createStateInInactivityLeak();
+    final UInt64 currentEpoch = spec.getCurrentEpoch(state);
+
+    final int slotsPerEpoch = spec.getSlotsPerEpoch(state.getSlot());
```

Comment:
i'd probably suggest moving these to member variables as they're basically intiialization.

The result would be we just have the actual test case in here, and it'll be a lot clearer.